integration-cli/docker_cli_build_test.go

```diff
@@ -3753,6 +3753,22 @@ ENV    abc 'yyy'
 RUN    [ $abc = \'yyy\' ]
 ENV    abc=
 RUN    [ "$abc" = "" ]
+ENV    abc=\$foo
```

Comment:
I would argue for a testcase for 

ENV abc=$foo
ENV abc $foo

which seems to gibe you full coverage of escaped, unescaped, both styles, $, ', and "